homeassistant/components/sensor/skybeacon.py

```diff
@@ -0,0 +1,181 @@
+    mon.start()
+
+
+class SkybeaconHumid(Entity):
+    """Representation of a humidity sensor."""
+
+    def __init__(self, name, mon):
+        """Initialize a sensor."""
+        self.mon = mon
+        self._name = name
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        return self.mon.humid
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit the value is expressed in."""
+        return "%"
+
+    @property
+    def state_attributes(self):
+        """Return the state attributes of the sensor."""
+        return {
+            ATTR_DEVICE: "SKYBEACON",
+            ATTR_MODEL: 1,
+        }
+
+class SkybeaconTemp(Entity):
+    """Representation of a temperature sensor."""
+
+    def __init__(self, name, mon):
+        """Initialize a sensor."""
+        self.mon = mon
+        self._name = name
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        return self.mon.temp
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit the value is expressed in."""
+        return TEMP_CELSIUS
+
+    @property
+    def state_attributes(self):
+        """Return the state attributes of the sensor."""
+        return {
+            ATTR_DEVICE: "SKYBEACON",
+            ATTR_MODEL: 1,
+        }
+
+class Monitor(threading.Thread):
+    """Connection handling."""
+
+    def __init__(self, hass, mac, name):
+        """Construct interface object."""
+        threading.Thread.__init__(self)
+        self.daemon = True
+        self.hass = hass
+        self.mac = mac
+        self.name = name
+        self.temp = STATE_UNKNOWN
+        self.humid = STATE_UNKNOWN
+
+    def run(self):
+        """Thread that keeps connection alive."""
+        adapter = pygatt.backends.GATTToolBackend()
+        while True:
+            try:
+                _LOGGER.info("connecting to %s", self.name)
+                # we need concurrent connect, so lets not reset the device
+                adapter.start(reset_on_start=False)
+                # seems only one connection can be initiated at a time
+                with CONNECT_LOCK:
+                    device = adapter.connect(self.mac,
+                                             CONNECT_TIMEOUT,
+                                             pygatt.BLEAddressType.random)
+                if SKIP_HANDLE_LOOKUP:
+                    # HACK: inject handle mapping collected offline
+                    # pylint: disable=protected-access
+                    device._characteristics[UUID(BLE_TEMP_UUID)] = CACHED_CHR
+                # magic: writing this makes device happy
+                device.char_write_handle(0x1b, bytearray([255]), False)
```

Comment:
do you mean standalone lib outside of homeassistant as such?

there is an issue that is somewhat related and that I already mentioned on the forum, but for convenience copypasting here:

There is a large issue  that you can't have two connection initiations at the same time on a given host. For example with gatttool I can't connect two devices on the same time. However, once connection is established, another connection can be made.

I tried to establish two connection attempts with bluepy-helper from Bluepy and it also failed. In fact with bluepy I can't seem to make 2 connections established at all.

I am not sure if this a bluez limitation OR my dongle (some CSR 4.0 usb leave-in) limitation. But looks like this is a very common dongle and it just break things. Right now I can't use bluetooth_le_tracker for the very same reason.

So what do we do? Provide a service that wraps and There is a large issue that you can't seem to have two connection initiations at the same time on a given host. For example with gatttool I can't connect two devices on the same time. However, once connection is established, another connection can be made.

I tried to establish two connection attempts with bluepy-helper from Bluepy and it also failed. In fact with bluepy I can't seem to make 2 connections established at all.

I am not sure if this a bluez limitation OR my dongle (some CSR 4.0 usb leave-in) limitation. But looks like this is a very common dongle and it just break things. Right now I can't use bluetooth_le_tracker for the very same reason.

So what do we do? Provide a service that wraps and serializes access to ble host controller?serialises access to ble host controller?